src/route-handlers/start-workflow/schemas/json-value-schema.ts

```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod';
+
+const literalSchema = z.union([z.string(), z.number(), z.boolean(), z.null()]);
+type Literal = z.infer<typeof literalSchema>;
+type Json = Literal | { [key: string]: Json } | Json[];
```

Comment:
nit: you can move these types to start-workflow.types.ts if you expect to reuse them.